### PR TITLE
remove abstract decorator from initialize in BaseEmbeddingEncoder

### LINKED ISSUE #3731: bug/unable to instantiate HuggingFaceEmbeddingEncoder from unstructured.embed.huggingface
**Describe the bug**

As of `0.16.0` I am unable to instantiate an instance of `HuggingFaceEmbeddingEncoder` from `unstructured.embed.huggingface` due to a missing abstract method (`initialize`)

**To Reproduce**
Provide a code snippet that reproduces the issue.

```python
from unstructured.embed.huggingface import (
    HuggingFaceEmbeddingConfig,
    HuggingFaceEmbeddingEncoder,
)

embedding_encoder = HuggingFaceEmbeddingEncoder(config=HuggingFaceEmbeddingConfig())
```

Returns `TypeError: Can't instantiate abstract class HuggingFaceEmbeddingEncoder without an implementation for abstract method 'initialize'`

For full example see: https://github.com/iterative/datachain/blob/main/examples/llm_and_nlp/unstructured-embeddings-gen.py 

**Expected behavior**

No error

**Screenshots**

N/A

**Environment Info**
Please run `python scripts/collect_env.py` and paste the output here. 
This will help us understand more about the environment in which the bug occurred.

I have seen this across Windows, Linux and Mac (https://github.com/iterative/datachain/actions/runs/11395986364)

**Additional context**

I opened to #3730 close this issue.

### PR DESCRIPTION
**Pull request by mattseddon**
Closes #3731. Please see the issue for details.

Comments inline.

### CODE REVIEW

File: unstructured/embed/interfaces.py [hunk 1]
```diff
@@ -15,7 +15,6 @@ class EmbeddingConfig(BaseModel):
 class BaseEmbeddingEncoder(ABC):
     config: EmbeddingConfig
 
-    @abstractmethod
```
Comment (mattseddon) [inline unstructured/embed/interfaces.py:18]: Matches the implementation of `BaseEmbeddingEncoder` in `unstructured_ingest`: https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/embed/interfaces.py#L16

**Comment by mattseddon** [comment]
@MKhalusova, can you help me get this reviewed?... please 🙏🏻🙏🏻🙏🏻. The DataChain unstructured examples ([here](https://github.com/iterative/datachain/blob/main/examples/llm_and_nlp/unstructured-embeddings-gen.py) and [here](https://github.com/iterative/datachain/blob/main/examples/llm_and_nlp/unstructured-summary-map.py)) are currently stuck on an older version of Unstructured because of #3731 and this change should close that issue.

**Comment by mattseddon** [comment]
@cragwolfe would this 👇🏻 be the correct format for the changelog entry? 


## 0.16.5-dev0

### Enhancements

### Features

### Fixes

* **Remove abstract decorator from initialize in BaseEmbeddingEncoder**

**Comment by scanny** [comment]
@rbiseck3 Hi Roman, can you take a look at this one-liner? I think you were closer to the choices on this one. I'm not sure whether this is the right fix or whether `HuggingFaceEmbeddingEncoder` needs to implement a trivial `.initialize()` method.

**Comment by mattseddon** [comment]
Can someone please take a look at this?

**Comment by rbiseck3** [comment]
@mattseddon all the embedder code was actually migrated to the new ingest repo, and this is no longer an issue there: [unstructured_ingest/embed/interfaces.py](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/embed/interfaces.py#L16). This code should be emitting a deprecation warning at the moment.

**Comment by mattseddon** [comment]
> @mattseddon all the embedder code was actually migrated to the new ingest repo, and this is no longer an issue there: [unstructured_ingest/embed/interfaces.py](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/embed/interfaces.py#L16). This code should be emitting a deprecation warning at the moment.

I didn't get the deprecation warning because the code won't run without this change and I had to revert to a previous version.

**Comment by mattseddon** [comment]
One other thing - the encoders have different function signatures (`list[dict]` vs `list[Element]`): 

https://github.com/mattseddon/unstructured/blob/df156ebe5ac4427ec7e2541e99cabb032801721d/unstructured/embed/huggingface.py#L54-L59

https://github.com/Unstructured-IO/unstructured-ingest/blob/e041b865b205358502e5d7f216d75b47219005ca/unstructured_ingest/embed/huggingface.py#L49-L52

**Pull request closed without merging**